### ISSUE TITLE
Fertile Planet

### ISSUE THREAD

**Issue by SaucyPigeon** [SaucyPigeon/Wild-Cultivation-Fan-Update#8]
Headshotkill 5 hours ago 
Hey back in 1.0 this mod conflicted with fertile planet leading to tropical jungle or swamp biomes to not spawn any vegetation. The problem was both fertile planet and wild cultivation added natural growing healroot to these biomes.
This problem was fixed by Rainbeau Flambe back then but it has now returned, I've already contacted the author of fertile planet 1.1 but he hasn't responded in the last couple days.

**Comment by SaucyPigeon** [comment]
When trying to load into a tropical rainforest:
```
Error in GenStep: System.ArgumentException: An item with the same key has already been added. Key: Plant_HealrootWild
  at System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) [0x000c1] in <567df3e0919241ba98db88bec4c6696f>:0 
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) [0x00000] in <567df3e0919241ba98db88bec4c6696f>:0 
  at RimWorld.BiomeDef.CachePlantCommonalitiesIfShould () [0x0003d] in <29684bdca4b441d38f64a3f064edc8ed>:0 
  at RimWorld.BiomeDef.CommonalityOfPlant (Verse.ThingDef plantDef) [0x00000] in <29684bdca4b441d38f64a3f064edc8ed>:0 
  at RimWorld.BiomeDef.get_AllWildPlants () [0x00031] in <29684bdca4b441d38f64a3f064edc8ed>:0 
  at RimWorld.WildPlantSpawner.CalculatePlantsWhichCanGrowAt (Verse.IntVec3 c, System.Collections.Generic.List`1[T] outPlants, System.Boolean cavePlants, System.Single plantDensity) [0x00054] in <29684bdca4b441d38f64a3f064edc8ed>:0 
  at RimWorld.WildPlantSpawner.CheckSpawnWildPlantAt (Verse.IntVec3 c, System.Single plantDensity, System.Single wholeMapNumDesiredPlants, System.Boolean setRandomGrowth) [0x00070] in <29684bdca4b441d38f64a3f064edc8ed>:0 
  at RimWorld.GenStep_Plants.Generate (Verse.Map map, Verse.GenStepParams parms) [0x0004b] in <29684bdca4b441d38f64a3f064edc8ed>:0 
  at Verse.MapGenerator.GenerateContentsIntoMap (System.Collections.Generic.IEnumerable`1[T] genStepDefs, Verse.Map map, System.Int32 seed) [0x000d5] in <29684bdca4b441d38f64a3f064edc8ed>:0 
Verse.Log:Error(String, Boolean)
Verse.MapGenerator:GenerateContentsIntoMap(IEnumerable`1, Map, Int32)
Verse.MapGenerator:GenerateMap(IntVec3, MapParent, MapGeneratorDef, IEnumerable`1, Action`1)
Verse.Game:InitNewGame()
Verse.<>c:<Start>b__1_2()
Verse.LongEventHandler:RunEventFromAnotherThread(Action)
Verse.<>c:<UpdateCurrentAsynchronousEvent>b__27_0()
System.Threading.ThreadHelper:ThreadStart_Context(Object)
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
System.Threading.ThreadHelper:ThreadStart()

```

**Comment by SaucyPigeon** [comment]
Both Fertile Planet and Wild Cultivation add healroot as a wild plant to tropical rainforest and tropical swamp.

**Comment by SaucyPigeon** [comment]
This error only occurs when Wild Cultivation is loaded before Fertile Planet, since Wild Cultivation removes the entry before adding it (while Fertile Planet adds it regardless). Can fix this issue by making Wild Cultivation load AFTER Fertile Planet.

**Comment by SaucyPigeon** [comment]
Fixed in version 1.0.5. Added specific patch for Fertile Planet to avoid duplicate key in dictionary.
